diff --color -uNr xscript-adcfdd7ac58dd222ccd5e9e4680e0e1872262006.orig/x-c1-boot.service xscript-adcfdd7ac58dd222ccd5e9e4680e0e1872262006/x-c1-boot.service
--- xscript-adcfdd7ac58dd222ccd5e9e4680e0e1872262006.orig/x-c1-boot.service	1969-12-31 18:00:00.000000000 -0600
+++ xscript-adcfdd7ac58dd222ccd5e9e4680e0e1872262006/x-c1-boot.service	2025-07-10 10:21:41.496371372 -0500
@@ -0,0 +1,11 @@
+[Unit]
+Description=Geekworm X-C1 assert BOOT GPIO17 HIGH at startup
+DefaultDependencies=no
+Before=basic.target
+
+[Service]
+Type=simple
+ExecStart=/usr/bin/gpioset --mode=signal 0 17=1
+
+[Install]
+WantedBy=basic.target
diff --color -uNr xscript-adcfdd7ac58dd222ccd5e9e4680e0e1872262006.orig/x-c1-pwr.service xscript-adcfdd7ac58dd222ccd5e9e4680e0e1872262006/x-c1-pwr.service
--- xscript-adcfdd7ac58dd222ccd5e9e4680e0e1872262006.orig/x-c1-pwr.service	2025-03-14 23:10:59.000000000 -0500
+++ xscript-adcfdd7ac58dd222ccd5e9e4680e0e1872262006/x-c1-pwr.service	2025-07-10 10:18:55.123452066 -0500
@@ -3,7 +3,7 @@
 
 [Service]
 Type=simple
-ExecStart=/usr/local/bin/xPWR.sh 0 4 17
+ExecStart=/usr/local/bin/xPWR.sh 0 4
 
 Restart=always
 User=root
diff --color -uNr xscript-adcfdd7ac58dd222ccd5e9e4680e0e1872262006.orig/xPWR.sh xscript-adcfdd7ac58dd222ccd5e9e4680e0e1872262006/xPWR.sh
--- xscript-adcfdd7ac58dd222ccd5e9e4680e0e1872262006.orig/xPWR.sh	2025-07-09 20:40:16.540294167 -0500
+++ xscript-adcfdd7ac58dd222ccd5e9e4680e0e1872262006/xPWR.sh	2025-07-10 10:18:38.593210609 -0500
@@ -3,14 +3,13 @@
 #  Use gpiod instead of obsolete interface, and suuports ubuntu 23.04 also
 
 # Make sure enough parameters are passed in  
-if [ "$#" -ne 3 ]; then  
-  echo "Usage: $0 <pwm_chip> <shutdown_pin> <boot_pin>"  
+if [ "$#" -ne 2 ]; then  
+  echo "Usage: $0 <pwm_chip> <shutdown_pin>"  
   exit 1  
 fi  
 
 GPIOCHIP=$1    
 SHUTDOWN=$2  
-BOOT=$3
 
 # Checks if the passed parameter is an integer
 re='^[0-9\.]+$'
@@ -22,16 +21,9 @@
    echo "error: shutdown_pin is not a number" >&2; exit 1
 fi
 
-if ! [[ $BOOT =~ $re ]] ; then
-   echo "error: button_pin is not a number" >&2; exit 1
-fi
-
 REBOOTPULSEMINIMUM=200  
 REBOOTPULSEMAXIMUM=600  
   
-# Initialize the BOOT pin to 1
-gpioset $GPIOCHIP $BOOT=1  
-
 function getTimestamp {
   python -c 'import time; print(int(time.time() * 1000))'
 }
